Menu: show text as tooltip for shrunk menus

bschwarzent opened the pull request:
400639

cguglielmo commented on eclipse-scout-core/src/action/Action.ts:
```diff
@@ -286,14 +290,14 @@ export class Action extends Widget implements ActionModel {
       tooltips.uninstall(this.$container);
     }
-    aria.description(this.$container, this.tooltipText);
+    aria.description(this.$container, this._tooltipText());
```
Maybe screen readers now read the same text twice because the aria-label and description contain the text. I am not sure about that, needs to be tested. Maybe we should only add the description if it differs from the text.

bschwarzent replied:
Correct. Aria attributes should not be affected by the shrinked state.

cguglielmo commented on eclipse-scout-core/src/action/Action.ts:
```diff
@@ -277,6 +277,10 @@ export class Action extends Widget implements ActionModel {
     this._updateTooltip();
   }
 
+  protected _tooltipText(): string {
```
Alternative name: _computeTooltipText

cguglielmo commented on eclipse-scout-core/src/menu/Menu.ts:
```diff
@@ -731,4 +732,15 @@ export class Menu extends Action implements MenuModel {
     return false;
   }
+
+  protected override _tooltipText(): string {
```
Can you move the declaration of textVisibleOrig  from Action to Menu? It seems to be at the wrong place.

cguglielmo commented on eclipse-scout-core/src/menu/Menu.ts:
```diff
@@ -731,4 +732,15 @@ export class Menu extends Action implements MenuModel {
+    return super._tooltipText();
+  }
+
+  protected _showTextAsTooltip(): boolean {
```
Please add a comment or jsdoc what the purpose of this function is (e..g same as in _renderText )

cguglielmo reviewed (commented):
Can you add Jasmine Specs?

I think I would move the code from MenuBoxLayout/MenuBarLayout to Action and Menu to simplify testing. 
Action.makeCompact/undoMakeCompact
Menu.shrink/undoShrink

Maybe keep suppressInvalidate in the Layout because it only needs to be suppressed because it is called while layouting.

bschwarzent committed:
Menu: show text as tooltip for shrunken menus

MenuBarLayout (used for menu bars) and MenuBoxLayout (used for the
desktop tool box) both reduce the size of menu items before moving
them to the ellipsis menu. One of these transformation is called
"shrinking". It only has an effect for menus with both an icon and a
text. When shrunk, the text is hidden and only the icon remains visible,
taking up much less space. However, this makes it less obvious for the
user what the menu does.

To improve this, the hidden text is now automatically shown as the
menu's tooltip. When the menu is un-shrunk, the tooltip is removed
again. If the menu already has a tooltip, both values are combined.

In the MenuBox (desktop tool box), we explicitly set all menus to be
shrinkable. This is a convenience to avoid having tp set all desktop
menus to shrinkable=true manually. It is only relevant on small screens
(e.g. mobile) and is therefore easily forgettable.

400639

bschwarzent commented:
Replaced by https://github.com/eclipse-scout/scout.rt/pull/1530
